src/Scenario/UserContactsScenario.php

```diff
@@ -61,42 +62,58 @@ public function parseNumbers(array $numbers, bool $withPhoto = false, bool $larg
             $counter = count($numbers);
             $models = [];
-            $cbGen = function ($phone) use (&$counter, &$models, $callback) {
-                return function (?UserInfoModel $userInfoModel) use (&$counter, $callback, &$models, $phone) {
-                    if ($userInfoModel) {
-                        $userInfoModel->phone = $phone;
-                        if (!$callback) {
-                            if ($userInfoModel->photo)
-                                file_put_contents(
-                                    $userInfoModel->phone.'.'.$userInfoModel->photo->format,
-                                    $userInfoModel->photo->bytes
-                                );
-                            echo "#################################\n";
-                            if ($userInfoModel->photo) {
-                                $userInfoModel->photo->bytes = 'HIDDEN';
+            $this->infoClient->reloadNumbers($numbers, function (ImportResult $result) use (&$models, $callback, $withPhoto, $largePhoto) {
+                $loadFlags = count($result->importedPhones);
+
+                foreach ($result->importedPhones as $importedPhone) {
+                    $this->infoClient->getContactByPhone($importedPhone, function (ContactUser $user) use (&$models, &$loadFlags, $callback, $withPhoto, $largePhoto) {
+                        $model = new UserInfoModel();
+                        $model->id = $user->getUserId();
+                        $model->phone = $user->getPhone();
+                        $model->langCode = $user->getLangCode();
+                        $model->firstName = $user->getFirstName();
+                        $model->lastName = $user->getLastName();
+                        $model->username = $user->getUsername();
+
+                        $this->infoClient->getFullUserInfo($user, $withPhoto, $largePhoto, function (UserInfoModel $fullModel) use ($model, &$models, $user, &$loadFlags, $callback) {
+                            $model->commonChatsCount = $fullModel->commonChatsCount;
+                            $model->status = $fullModel->status;
+                            $model->bio = $fullModel->bio;
+
+                            $models[$user->getUserId()] = $model;
+                            $loadFlags--;
+
+                            if ($loadFlags == 0) {
+                                $this->reloadUsersInfo($models, $callback);
                             }
-                            Logger::log(__CLASS__, print_r($userInfoModel, true));
-                        } else {
-                            $counter--;
-                            $models[] = $userInfoModel;
-                        }
-                    } else {
-                        $counter--;
-                    }
+                        });
+                    });
+                    sleep(2);
+                }
+            });
+        };
```

Comment:
Alghorithm should be following:
1) reloadContacts
2) rememberedContacts = getCurrentContacts (get all possible info, except first/last names)
3) cleanContacts
4) foreach contact in rememberedContacts { getUserFull(contact) -> setuo last\first names }